src/dto/variation-dto.ts

```diff
@@ -1,11 +1,12 @@
+import { IValue } from '../eppo_value';
+
 export interface IShardRange {
   start: number;
   end: number;
 }
 
 export interface IVariation {
   name: string;
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  value: any;
+  typedValue: IValue;
```

Comment:
`IValue` only supports `boolean`, `number` and `string`. But the typedValue can also be a json right?

Reply:
Yes in that case it will be a `string`; in this PR `IValue` is a type so that json deserialization works and `EppoValue` is the container for the various values.

What I am missing from being able to unify them is something like `UnmarshalJSON` in golang which is called when an object is being json deserialized.

https://github.com/Eppo-exp/golang-sdk/blob/main/eppoclient/value.go

Do you have an idea for how to accomplish this in javascript?